x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.test.ts

```diff
@@ -0,0 +1,291 @@
+      fetchMock.mockImplementation(() => mockSignalIndex);
+    });
+    test('check parameter url', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await getSignalIndex({ signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/index', {
+        method: 'GET',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess(null, fetchMock);
+      const signalsResp = await getSignalIndex({
+        signal: abortCtrl.signal,
+      });
+      expect(signalsResp).toEqual(mockSignalIndex);
+    });
+    test('unhappy path', async () => {
+      mockfetchError();
+      try {
+        await getSignalIndex({ signal: abortCtrl.signal });
+      } catch (exp) {
+        expect(exp).toBeInstanceOf(SignalIndexError);
+        expect(exp.message).toEqual('super mega error, it is not that bad');
+      }
+    });
+  });
+
+  describe('getUserPrivilege', () => {
+    beforeEach(() => {
+      mockKibanaServices.mockClear();
+      fetchMock.mockClear();
+      fetchMock.mockImplementation(() => ({
+        response: {
+          ok: true,
+          message: 'success',
+          text: 'success',
+        },
+        body: mockUserPrivilege,
+      }));
+    });
+    test('check parameter url', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await getUserPrivilege({ signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/privileges', {
+        asResponse: true,
+        method: 'GET',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess(mockUserPrivilege);
+      const signalsResp = await getUserPrivilege({
+        signal: abortCtrl.signal,
+      });
+      expect(signalsResp).toEqual(mockUserPrivilege);
+    });
+    test('unhappy path', async () => {
+      mockfetchError();
+      try {
+        await getUserPrivilege({ signal: abortCtrl.signal });
+      } catch (exp) {
+        expect(exp).toBeInstanceOf(ToasterErrors);
+        expect(exp.message).toEqual('super mega error, it is not that bad');
+      }
+    });
+  });
+
+  describe('createSignalIndex', () => {
+    beforeEach(() => {
+      mockKibanaServices.mockClear();
+      fetchMock.mockClear();
+      fetchMock.mockImplementation(() => mockSignalIndex);
+    });
+    test('check parameter url', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await createSignalIndex({ signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/index', {
+        method: 'POST',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess(null, fetchMock);
+      const signalsResp = await createSignalIndex({
+        signal: abortCtrl.signal,
+      });
+      expect(signalsResp).toEqual(mockSignalIndex);
+    });
+    test('unhappy path', async () => {
```

Comment:
The other true false positive: a bad response does not cause an error to be thrown (at least in the test).

Taking a step back, we have some drift in the way we handle the request/response in both the get/createSignalIndex functions. This is because you want to throw a specific error, right? It might be worth adapting those functions to use `asResponse` and `throwIfNotOk`, and catch/re-throw that with your specific errors.